### DISCUSSION TITLE
Roadmap Progress - July and August 2023 updates

### DISCUSSION THREAD

**Discussion by pchaurasia14** [dotnet/wpf#8225] [Announcements]
Below are the updates on 2023 Roadmap deliverables - 

## Modernization

| Area  | Update |
| ------------- |:-------------|
| Win11 Theming | We are currently awaiting the finalization of some cross-team dependencies by our partner teams before we can make further progress on this. We are working towards drafting the spec document for this. We will share more details as soon as we have them. |
| New control - FolderBrowserDialog | Feature [merge](https://github.com/dotnet/wpf/pull/7244) is **completed** 🎉🎉.<br/> [Sample application](https://github.com/microsoft/WPF-Samples/pull/500) and [tests](https://github.com/dotnet/wpf-test/pull/130) are available as well. |
| Nullability annotations | Based on community [feedback](https://github.com/dotnet/wpf/discussions/7555#discussioncomment-5053915), the plan for Nullability annotations support is available [here](https://github.com/dotnet/wpf/issues/7563).<br/> Ready for community contributions. |
| Hardware acceleration in RDP | We are happy to announce that we've also merged the [PR](https://github.com/dotnet/wpf/pull/7684) for enabling hardware acceleration in RDP. This would enable applications on RDP to take advantage of hardware acceleration capability (improved frames/second support, smoother animations, use of video memory, etc.) if the underlying system supports it. Our testing on [sample applications](https://github.com/microsoft/WPF-Samples/tree/main/Animation/AnimationExamples) which uses animations showed an improvement of 10-30% sustained frame rate when running over RDP. <br/><br/>To enable the feature, you need to specify an appcontext switch(`System.Windows.Media.IsHardwareAccelerationEnabledForRdp`) in either .csproj or *.runtimeconfig.json . Do try this feature out and provide us with your feedback.|

---

## Infrastructure upgrades

| Area  | Update |
| ------------- |:-------------|
| Test Automation | We have enabled [unit testing](https://github.com/dotnet/wpf/pull/8013) as well, using build pipelines that are already capable of executing DRTs (_DRT means "developer regression tests" -- these are tests that check the basics are not broken_) [example](https://dev.azure.com/dnceng-public/public/_build/results?buildId=404572&view=ms.vss-test-web.build-test-results-tab), microsuites, and feature tests. <br/><br/>Special thanks to @hughbe for starting the process of adding unit tests to the repository. Moving forward, we would also concentrate on raising the project's unit test count to enable improved coverage and, as a result, more confidence in problem fixes. <br/> These tests runs will be scheduled at regular intervals rather than on each build because it takes a few days to execute all of the microsuites and feature tests. Unit tests and DRTs will continue to run on every build. |
| Test Migration | Majority for feature tests and microsuites are now available in test repo (~ 80% completed) Few areas that are remaining have a higher amount on infrastructure dependency, hence we would migrate them as and when we incorporate changes to those feature areas. <br/><br/>- UI Automation (~ 95) - [PR](https://github.com/dotnet/wpf-test/pull/183) <br/> - FlowLayout (~ 336) - [PR](https://github.com/dotnet/wpf-test/pull/148) <br/> - WindowsFormIntegration -[PR](https://github.com/dotnet/wpf-test/pull/157) <br/> - Editing [PR](https://github.com/dotnet/wpf-test/pull/163)  <br/> - 2D (in progress) <br/> - 3D (in progress) <br><br/> _Number of tests are indicative of total number of tests made available (including disabled tests)_<br/> You can follow our progress overall [in this project](https://github.com/orgs/dotnet/projects/145). |

---

## Issues/ PRs

| Area  | Update |
| ------------- |:-------------|
| PRs | Following community PRs have been merged for July and August - <br/>  <br/> - OpenFolderDialog [#7244](https://github.com/dotnet/wpf/pull/7244) (Thanks @miloush)<br/> - Common file dialogs new properties [#7916](https://github.com/dotnet/wpf/pull/7916) (Thanks @miloush)<br/> - Replace String.CompareOrdinal to string.Equals Final Part [#8042](https://github.com/dotnet/wpf/pull/8042) (Thanks @lindexi)<br/> - Fix build when using Razor source generator [#6535](https://github.com/dotnet/wpf/pull/6535) (Thanks @ThomasGoulet73)<br/> - Cleanup System.Windows.Markup code [#207](https://github.com/dotnet/wpf/pull/207) (Thanks @hughbe)<br/>- Avoid crashing if temporary file cannot be deleted [#7734](https://github.com/dotnet/wpf/pull/7734) (Thanks @miloush)<br/> - Add Windows.Markup tests [#138](https://github.com/dotnet/wpf/pull/138) (Thanks @hughbe) <br/> - Fix bug where x:Static argument is treated as member of 'SystemParameters' by mistake [#680](https://github.com/dotnet/wpf/pull/680) (Thanks @KodamaSakuno) <br/> - Preventing focus steal if parent window/application is not active [#1169](https://github.com/dotnet/wpf/pull/1169) (Thanks @batzen) <br/> - Reduce allocations when tracing routed events [#6700](https://github.com/dotnet/wpf/pull/6700) (Thanks @bgrainger) <br/> - Use ArgumentNullException.ThrowIfNull in PresentationCore [#8126](https://github.com/dotnet/wpf/pull/8126) (Thanks @halgab) <br/> - Use ArgumentNullException.ThrowIfNull in WindowsBase [#8125](https://github.com/dotnet/wpf/pull/8125) (Thanks @halgab) <br/> - Use ArgumentNullException.ThrowIfNull in ReachFramework [#8121](https://github.com/dotnet/wpf/pull/8121) (Thanks @halgab) <br/> - Use ArgumentNullException.ThrowIfNull in PresentationFramework [#8092](https://github.com/dotnet/wpf/pull/8092) (Thanks @halgab) <br/> <br/>    We are very thankful to all the contributors who have helped improve WPF.<br/> We are aiming to reduce the total turnaround time for issues and PRs.  |
| Issues | We were able to address the following issues - <br/> <br/> - Guard with null checks in adorners [#8086](https://github.com/dotnet/wpf/pull/8086). This was reported via internal compat testing.<br/>- Fixing the AssemblyName in temporary project file for GenerateTemporaryTargetAssembly task [#7557](https://github.com/dotnet/wpf/pull/7557) |

---

## Misc

| Area  | Update |
| ------------- |:-------------|
| Accessibility improvements | No major changes to report |
| ARM64 | Not yet started |

**Comment by psmulovics** [comment]
hurray!
